filters/auth/tokeninfo.go

```diff
@@ -85,12 +90,13 @@ func NewOAuthTokeninfoAllKV(OAuthTokeninfoURL string, OAuthTokeninfoTimeout time
 // to validate authorization for requests. Current implementation uses
 // Bearer tokens to authorize requests and checks that the token
 // contains at least one key value pair provided.
-func NewOAuthTokeninfoAnyKV(OAuthTokeninfoURL string, OAuthTokeninfoTimeout time.Duration) filters.Spec {
+func NewOAuthTokeninfoAnyKV(OAuthTokeninfoURL string, OAuthTokeninfoTimeout time.Duration, tracer opentracing.Tracer) filters.Spec {
```

Comment:
changing the signature of these initializers makes it hard for applications using Skipper as a library to migrate to the new version. In other parts of the code base, we tend to use an Options structure to flexibly extend the parameters of exported initializers. I'd suggest to stick to this approach. This would become something like this:

```
// extend the existing options
type TokeninfoOptions struct {
	OAuthTokeninfoURL string
	OAuthTokeninfoTimeout time.Duration
	Tracer opentracing.Tracer
	// ...the existing fields
}

func OAuthTokeninfoAnyKVWitOptions(o TokeninfoOptions) filters.Spec {
	return &tokeninfoSpec{
		typ: checkOAuthTokeninfoAnyKV,
		options: o,
	}
}

func NewOAuthTokeninfoAnyKV(OAuthTokeninfoURL string, OAuthTokeninfoTimeout time.Duration) {
	return OAuthTokeninfoAnyKVWitOptions(TokeninfoOptions{
		OAuthTokeninfoURL string
		OAuthTokeninfoTimeout time.Duration
		Tracer opentracing.Tracer
	})
}
```

there are other similar occurences, please change them, too

Reply:
fixed